### ISSUE TITLE
Companion to detect if REM is on

### ISSUE THREAD

**Issue by hadphild** [bitfocus/companion-module-bmd-hyperdeck#115]
Companion to detect if REM is on the hyper deck if not then inform user.

This could be build into the button presets for REM in Red when hitting record.

**Comment by Julusian** [comment]
I'm pretty sure there is a feedback for this already.

But perhaps it should be possible to tell the module to affect the status of the module if it is not enabled. The status of the module could be put into the 'warning' state when remote isn't enabled.
This would need to be opt-out in the config as I can imagine some scenarios where this is expected or intended.

**Comment by hadphild** [comment]
You are right. the feedback is there. It just does not show on the instance.

**Comment by sphlabs** [comment]
I don't believe that this should affect the instance status, which is there to convey the status of the module's connection to the end device. In this case the connection is established and has no issues, so the status should show OK.

Whether REM is on or off is a function of the Hyperdeck's state, not the Companion module's state. If that is something you want to know about, there is feedback and a variable for that purpose (and even an action to change it).

**Comment by Julusian** [comment]
But isn't REM not being enabled the same thing as failing authentication? It essentially puts the connection into readonly mode until corrected.
And it would definitely help some users if the status was a warning, rather than relying on them knowing that they need to enable REM. This can be done with an action, or on the hardware, but it still requires the user to be aware that they might need to fix this.
I have forgotten to check whether REM is enabled on some hired hyperdecks before, I only realised once an action didn't work.
So I do think it should by default make the status a warning, with a toggle to opt out.

**Comment by sphlabs** [comment]
No, not at all the same thing at all. Even with REM disabled, Companion is still able to make a fully functional connection to the deck. There is nothing at all wrong with *the instance connection* that would warrant a Warning state.

Having REM disabled on a Hyperdeck is not really any different to having FTB enabled on an ATEM–*'nothing else works' until it is disabled*–or Mute All on a sound desk. The fact that REM can be turned on or off by Companion means that you don't even need physical access to the deck to change state (and further reinforces the fact that the Companion connection is just fine).

True, a user may be tripped up if they don't know about REM, but that's part of the learning curve for every piece of production hardware out there, and it is not the place of the instance page to convey the device status; it is only for connection status, and should only show a warning if there is a legitimate fault with the module's connection to the device, which in this case there isn't.

There isn't even a suitable status enum value that would apply here. It is certainly none of `BadConfig, Connecting, ConnectionFailure, Disconnected, Ok, UnknownError, UnknownWarning`

**Comment by Julusian** [comment]
Resolved in v3.0.0

There is a config option to opt into giving the connection a warning state when remote is disabled
